Mlem/App/Views/Shared/Search/VisitHistory.swift

```diff
@@ -104,6 +117,16 @@ class VisitHistory {
         }
     }
     
+    private func removeValue<T: Equatable>(
+        _ value: T,
+        from dict: inout [VisitContext: [VisitRecord<T>]],
+        context: VisitContext
+    ) {
+        if let index = dict[context, default: []].firstIndex(where: { $0.value == value }) {
+            dict[context, default: []].remove(at: index)
+        }
+    }
+    
     func clear() {
         communityRecords = [:]
```

Comment:
Oops, just realised I must have left the other dictionaries out of this method in my original PR
```suggestion
        communityRecords = [:]
        personRecords = [:]
        instanceRecords = [:]
```